### PR TITLE
#151991945 Registered users should be able to view a "game log" of past games

### PR DESCRIPTION
**Pull request by ChidinmaOrajiaku**
#### What does this PR do?
 - Creates "getGameLog" and "getDonations" routes
-  Creates dashboard controller
- Creates dashboard user interface
- Adds dashboard controller to app views

#### Description of Task to be completed?
- It allows users to view a dashboard containing past games, leaderboard and donations

#### How should this be manually tested?
-Navigate to https://mont-blanc-cfh.herokuapp.com/
or test locally with npm start:dev

#### Any background context you want to provide?
None

#### What are the relevant pivotal tracker stories?
#151991945 Registered users should be able to view a a "game log" of past games

#### Screenshots (if appropriate)
image
image
image

#### Questions:
- None

### CODE REVIEW

File: app/controllers/games.js [hunk 1]
```diff
@@ -98,3 +98,37 @@ export const getLeaderboard = (req, res) => {
...
+          gameID: { $first: '$gameID' },
+          gameRound: { $first: '$gameRound' },
+          gameWinner: { $first: '$gameWinner' },
+          gamePlayers: { $first: '$gamePlayers' }
+        }
+      }
+    ])
+      .exec((err, gameLog) => {
+        if (err) {
+          return res.status(500).send({ err });
```
Comment (temilaj) [inline app/controllers/games.js]: should we just send the server error to the client? or do we want to make sure we send a user friendly error?

File: test/user/donations.js [hunk 2]
```diff
@@ -0,0 +1,69 @@
...
+  expiresIn: '0.001s'
+});
+const User = mongoose.model('User');
+
+// delete all records in User model
+User.collection.drop();
+
+const userDetails = {
+  name: 'Oluwadunsin Oyebiyi',
+  password: 'password',
```
Comment (temilaj) [inline test/user/donations.js:28]: Where is the test that actually asserts for the feature we have worked on?

Comment (ChidinmaOrajiaku) [reply test/user/donations.js:28]: Please scroll down to view the tests

File: app/controllers/games.js (hunk 1 above)
Comment (ChidinmaOrajiaku) [reply app/controllers/games.js]: Feedback will be implemented immediately

File: public/js/controllers/dashboard.js [hunk 3]
```diff
@@ -0,0 +1,112 @@
...
+        $scope.donationsCount = range;
+      });
+    };
+
+    $scope.getGameLog = (token) => {
+      token = localStorage.getItem('token');
+      $http.get(`/api/v1/games/history/${token}`).then((user) => {
+        $scope.gamesData = user.data;
+        const range = [];
+        for (let index = 1; index < (user.data).length + 1; index += 1) {
```
Comment (temilaj) [inline public/js/controllers/dashboard.js:34]: It's advisable not to use iterators (loops), but instead, Prefer JavaScript’s higher-order functions.

**Review by temilaj — CHANGES_REQUESTED** [review]
Good job improving your test suite. work on the other comment I made

File: test/user/donations.js (hunk 2 above)
Comment (temilaj) [reply test/user/donations.js:28]: Do you have any test for "users should be able to view a "game log" of past games"?

**Pull request closed without merging**